docs: explain Renovate config for GitHub Merge Queue (#20452)
renovatebot/renovate: docs/usage/configuration-options.md

Please check platform specific docs for version requirements.

> To learn how to use GitHub's Merge Queue feature with Renovate, read the [Key Concepts page, GitHub Merge Queue](./key-concepts/automerge.md#github-merge-queue) page.

Author: ```suggestion
To learn how to use GitHub's Merge Queue feature with Renovate, read our [Key Concepts, Automerge, GitHub Merge Queue](./key-concepts/automerge.md#github-merge-queue) docs.
```
